src/storage.ts

```diff
@@ -62,6 +62,47 @@ export function getAssetUrl(path: string, base: URL | string): string {
   return `${assetUrlBase.replace(/\/+$/, "")}/${path.replace(/^\/+/, "")}`;
 }
 
+/**
+ * GetRegion from S3 Endpoint URL
+ *
+ * @example
+ * // AWS S3
+ * getFormattedEndpoint('https://bucket.s3.us-east-1.amazonaws.com', undefined) // returns 'us-east-1'
+ *
+ * // S3-compatible storage (e.g., MinIO)
+ * getFormattedEndpoint('http://minio:9000', undefined) // returns undefined
+ *
+ * @param envEndpoint
+ * @param envRegion
+ * @returns {string|undefined} The region string if detected or provided, undefined otherwise
+ */
+const getFormattedEndpoint = (envEndpoint?: string, envRegion?: string) => {
```

Comment:
I think the name of this function is somewhat misleading? How about `getDefaultRegion()`?